update.c

```diff
@@ -187,6 +191,89 @@ do_run_dialog (GtkListStore *list_store,
 }
 
 static void
+update_background_image_path (XdgDirEntry *old_entries,
+                              XdgDirEntry *new_entries)
+{
+  GSettings *settings = NULL;
+
+  XdgDirEntry *old_entry = NULL;
+  XdgDirEntry *new_entry = NULL;
+
+  char *old_uri = NULL;
+  char *new_uri, *new_path, *old_path;
+  char **split_path = NULL;
+
+  settings = g_settings_new (WP_PATH_ID);
+
+  old_uri = g_settings_get_string (settings, WP_URI_KEY);
+  if (old_uri == NULL)
+    {
+      g_warning ("No old image URI set. Ignoring background image change.");
```

Comment:
No need for a warning here. It's not an inconsistent state not to have this set.